### PR TITLE
Tg 2 67 node express set project route

### PR DESCRIPTION
**Pull request by salitordjman**
[](url)
[](url)

### CODE REVIEW

**Comment by salitordjman** [comment]
[TG2-64](https://appleseedsstudents.atlassian.net/browse/TG2-64)
[TG2-69](https://appleseedsstudents.atlassian.net/browse/TG2-69)
[TG2-70](https://appleseedsstudents.atlassian.net/browse/TG2-70)
[TG2-67](https://appleseedsstudents.atlassian.net/browse/TG2-67)

File: backend/src/controllers/projects.controller.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    res.status(200).send(projectFilter);
+  } catch (error) {
+    res.status(400).send(error);
+  }
+}
+
+async function createProject(req: Request, res: Response) {
+  try {
+    const { name, description, goal, status: boolean } = req.body;
+
```
Comment (Toam10) [inline backend/src/controllers/projects.controller.ts:34]: please remove the boolean

File: backend/src/controllers/projects.controller.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    if (projectDetail.length === 0) return res.status(404).send('Not found a Project');
+
+    res.status(200).send(projectDetail);
+  } catch (error) {
+    res.status(400).send(error);
+  }
+}
+
+function getAllProjectDetails(req: Request, res: Response) {
+  try {
```
Comment (Toam10) [inline backend/src/controllers/projects.controller.ts:19]: change to async

File: backend/src/controllers/projects.controller.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { Request, Response } from 'express';
+import { Project, projects } from '../fakeData/project.fakeData';
+
+async function getProjectDetails(req: Request, res: Response) {
+  try {
+    const projectDetail = await projects.filter((project) => project.id === req.params.id);
+
```
Comment (Toam10) [inline backend/src/controllers/projects.controller.ts:7]: remove the await

Comment (Toam10) [reply backend/src/controllers/projects.controller.ts:7]: change to fine method

File: backend/src/controllers/projects.controller.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    // const projectDetail =await projects.findById(req.params.id);
+
+    if (projectDetail.length === 0) return res.status(404).send('Not found a Project');
+
```
Comment (Toam10) [inline backend/src/controllers/projects.controller.ts:11]: please throw error and res.send this message

**Review by Toam10 — APPROVED** [review]

**Pull request merged**